### PR TITLE
1384 moves basic table to library

### PR DESCRIPTION
**Pull request by ricardovdheijden**
#1384 
Breaking change

Changes:
- WfoBasicTable replaced with WfoTable -- Most basic version of a table in the project
- WfoTableWithFilter replaced with WfoAdvancedTable -- A more advanced version including filtering and column selection
- Removes WfoDataGridTable
- Adds WfoGroupedTable -- A new component to be used to make nested groups based on the input data in the table:

image

The new WfoTable does not have the EuiTable as base table anymore.

### CODE REVIEW

**Comment by ricardovdheijden** [comment]
Draft: Need to handle some todo's in the code

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx [hunk 1]
```diff
@@ -70,82 +73,78 @@ export const WfoRelatedSubscriptions = ({
...
+            label: t('startDate'),
             width: '100',
-            render: (value) => parseDateToLocaleDateString(parseDate(value)),
-            filterable: false,
+            renderData: (value) =>
+                parseDateToLocaleDateString(parseDate(value)),
         },
     };
 
+    const onUpdatePageIndex =
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:129]: Small nitpick but I would rename this to onChangeItemsPerPage right away to you pass likt that in line 140 right away

Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:129]:
```suggestion
    const onChangeItemsPerPage =
```

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx [hunk 2]
```diff
@@ -70,82 +73,78 @@ export const WfoRelatedSubscriptions = ({
...
+        getPageIndexChangeHandler<RelatedSubscription>(setDataDisplayParam);
+    const onUpdatePageSize =
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:131]:
```suggestion
    const onChangeItemsPerPage = 
```

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx [hunk 3]
```diff
@@ -70,82 +73,78 @@ export const WfoRelatedSubscriptions = ({
...
+        getPageSizeChangeHandler<RelatedSubscription>(setDataDisplayParam);
+
     const pagination: Pagination = {
         pageSize: dataDisplayParams.pageSize,
         pageIndex: dataDisplayParams.pageIndex,
-        pageSizeOptions: DEFAULT_PAGE_SIZES,
         totalItemCount: relatedSubscriptionsPageInfo?.totalItems ?? 0,
+        pageSizeOptions: DEFAULT_PAGE_SIZES,
+        onChangeItemsPerPage: onUpdatePageSize,
+        onChangePage: onUpdatePageIndex,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:140]:
```suggestion
        onChangeItemsPerPage,
        onChangePage
```

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/WfoAdvancedTable.tsx [hunk 4]
```diff
@@ -120,18 +98,17 @@ export const WfoTableWithFilter = <T extends object>({
...
-            ...(detailModal ? detailsIconColumn : []),
-        };
+    const tableColumnsWithControlColumns: WfoAdvancedTableColumnConfig<T> = {
+        ...(detailModal && detailsIconColumn),
+        ...tableColumnConfig,
+    };
+
+    const tableSettingsColumns: TableSettingsColumnConfig<T>[] = Object.entries(
+        tableColumnConfig,
+    ).map((keyValuePair) => {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/WfoAdvancedTable.tsx:108]: can't you destructure here? Something like:

```suggestion
    ).map(([key, value]) => {
```

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    ColumnType,
+    WfoTableControlColumnConfigItem,
+} from '@/components/WfoTable/WfoTable';
+import { getTypedFieldFromObject } from '@/utils';
+
+export const getRowDetailData = <T extends object>(
+    selectedDataForDetailModal: T | undefined,
+    tableColumnConfig: WfoAdvancedTableColumnConfig<T>,
+) => {
+    if (!selectedDataForDetailModal) {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx:17]: What is the use case for this? Why just not call it if you don't want to use it in a detail modal?

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    const tableColumnConfigEntries: [
+        string,
+        (
+            | WfoTableControlColumnConfigItem<T>
+            | WfoAdvancedTableDataColumnConfigItem<T, keyof T>
+        ),
+    ][] = Object.entries(tableColumnConfig);
+
+    const dataColumnEntries = tableColumnConfigEntries.filter(
+        (tableColumnConfigEntry) =>
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx:30]: I prefer to do something more descriptive like this so you don't have to do the access by [1] in line 31
```suggestion
        ([_, columnConfig]) =>
```

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            tableColumnConfigEntry[1].columnType === ColumnType.DATA,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx:31]:
```suggestion
            columnConfif.columnType === ColumnType.DATA,
```

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoTable/WfoGroupedTable/WfoExpandedGroupRow.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                data={data}
+                columnConfig={columnConfig}
+                groupNameLabel={groupNameLabel}
+                nestingLevel={nestingLevel}
+                onExpandRowChange={onExpandRowChange}
+            />
+        );
+    },
+);
+WfoExpandedGroupRow.displayName = 'WfoExpandedGroupRow';
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WfoTable/WfoTable/WfoGroupedTable/WfoExpandedGroupRow.tsx:71]: What does this do?

**Review by DutchBen — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx (hunk 6 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx:30]: I prefer this too, but the current linting rules do not allow this. It still sees `_` as an unused variable

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx (hunk 1 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:129]: done

Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:129]: done

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx (hunk 2 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:131]: done

File: packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx (hunk 3 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoSubscription/WfoRelatedSubscriptions.tsx:140]: done

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/WfoAdvancedTable.tsx (hunk 4 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/WfoAdvancedTable.tsx:108]: done

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx (hunk 5 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx:17]: Good one, selectedDataForDetailModal is now only `T` and moved the undefined check to WfoAdvancedTable component

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx (hunk 7 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/getRowDetailData.tsx:31]: done

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoTable/WfoExpandedRow.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            value,
+        ])
+        .filter(
+            ([key]) =>
+                key ===
+                (
+                    rowData[rowExpandingConfiguration.uniqueRowId] as string
+                ).toLowerCase(),
+        )
+        .map((keyValue) => keyValue[1]);
```
[the remark this answers is not in the document]
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoTable/WfoTable/WfoExpandedRow.tsx:32]: done

File: packages/orchestrator-ui-components/src/components/WfoTable/WfoTable/WfoGroupedTable/WfoExpandedGroupRow.tsx (hunk 8 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoTable/WfoTable/WfoGroupedTable/WfoExpandedGroupRow.tsx:71]: When applying the `React.forwardRef` we need to specify the name of the component in the prop `displayName`

File: packages/orchestrator-ui-components/src/components/WfoSubscriptionsList/WfoSubscriptionsList.tsx [hunk 10]
```diff
@@ -69,91 +70,88 @@ export const WfoSubscriptionsList: FC<WfoSubscriptionsListProps> = ({
...
-            name: t('metadata'),
-            render: (value) => <WfoInlineJson data={value} />,
-            renderDetails: (value) =>
-                value && <WfoJsonCodeBlock data={value} isBasicStyle />,
-        },
-    };
+    const tableColumnConfig: WfoAdvancedTableColumnConfig<SubscriptionListItem> =
+        {
+            subscriptionId: {
+                columnType: ColumnType.DATA,
```
Comment (Georgi2704) [inline packages/orchestrator-ui-components/src/components/WfoSubscriptionsList/WfoSubscriptionsList.tsx:81]: This seems a bit repeated. What about making the default  ColumnType = DATA ?

**Review by Georgi2704 — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WfoSubscriptionsList/WfoSubscriptionsList.tsx (hunk 10 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WfoSubscriptionsList/WfoSubscriptionsList.tsx:81]: In the type `WfoAdvancedTableColumnConfig` the `columnType` prop is used to discriminate between the 2 types. As of now it does not seem to be possible to make it optional. Making it optional allowes both props renderData and renderControl to be part of the configuartion

**Pull request merged**